packages/common/src/cis2/util.ts

```diff
@@ -0,0 +1,504 @@
+     */
+    export type UpdateOperator = {
+        /** The type of the update */
+        type: 'add' | 'remove';
+        /** The address be used for the operator update */
+        address: Address;
+    };
+
+    /**
+     * Metadata necessary for CIS-2 transactions
+     */
+    export type TransactionMetadata = {
+        /** Amount (in microCCD) to inlude in the transaction. Defaults to 0n */
+        amount?: bigint;
+        /** The sender address of the transaction */
+        senderAddress: HexString;
+        /** Expiry date of the transaction. Defaults to 5 minutes in the future */
+        expiry?: Date;
+        /** Max energy to be used for the transaction */
+        energy: bigint;
+    };
+
+    /**
+     * Metadata necessary for creating a {@link UpdateTransaction}
+     */
+    export type CreateTransactionMetadata = Pick<
+        TransactionMetadata,
+        'amount' | 'energy'
+    >;
+
+    /**
+     * Data needed for CIS-2 "balanceOf" query.
+     */
+    export type BalanceOfQuery = {
+        /** The ID of the token to query */
+        tokenId: HexString;
+        /** The address to query balance for */
+        address: Address;
+    };
+
+    /**
+     * Structure for holding metadata URL reponse from tokenMetadata query.
+     */
+    export type MetadataUrl = {
+        /** The URL of the metadata */
+        url: string;
+        /** An optional checksum for the URL */
+        hash?: HexString;
+    };
+
+    /**
+     * Data needed for CIS-2 "operatorOf" query.
+     */
+    export type OperatorOfQuery = {
+        /** The owner address for the query */
+        owner: Address;
+        /** The address to check whether it is an operator of `owner` */
+        address: Address;
+    };
+
+    /**
+     * An update transaction without header. This is useful for sending through a wallet, which supplies the header information.
+     */
+    export type UpdateTransaction = {
+        /** The type of the transaction, which will always be of type {@link AccountTransactionType.Update} */
+        type: AccountTransactionType.Update;
+        /** The payload of the transaction, which will always be of type {@link UpdateContractPayload} */
+        payload: UpdateContractPayload;
+        parameter: {
+            /** Hex encoded parameter for the update */
+            hex: HexString;
+            /** JSON representation of the parameter to be used with the corresponding contract schema */
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            json: any;
+        };
+        schema: {
+            /** Base64 encoded schema for the parameter type */
+            value: Base64String;
+            /** Type of the schema. This is always of type "parameter" */
+            type: 'parameter';
+        };
+    };
+
+    /**
+     * A function processing a {@link UpdateTransaction}
+     */
+    export type ProcessTransactionFunction<R> = (
```

Comment:
I don't understand what the purpose of this is? :thinking:
What is the benefit of using this, instead of making the `createUpdateTransaction` function public?